### PR TITLE
update the err of hostPorts in psp

### PR DESCRIPTION
**Pull request by xilabao**
**What this PR does / why we need it**:
change `Allowed ports: [{8000 8080}]` to `Allowed ports: [8000-8080]`

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cblecker** [comment]
@k8s-bot ok to test

**Comment by php-coder** [comment]
WDYT about explicitly specifying min/max port and not exposing structure names. This also makes a messages cleaner and shorter. For example: `Allowed ports: range between 8000 and 8080` is more readable IMHO.

**Comment by xilabao** [comment]
@php-coder hostports may have multi ranges. what about `Allowed ports: range between {8000-8080,7000-7080}`? ptal

File: pkg/security/podsecuritypolicy/provider.go [hunk 1]
```diff
@@ -339,3 +340,15 @@ func (s *simpleProvider) isValidHostPort(port int) bool {
 func (s *simpleProvider) GetPSPName() string {
 	return s.psp.Name
 }
+
+func hostPortRangeToString(ranges []extensions.HostPortRange) string {
```
Comment (php-coder) [inline pkg/security/podsecuritypolicy/provider.go]: Neat: `hostPortRangeToString` -> `hostPortRangesToString`

File: pkg/security/podsecuritypolicy/provider.go [hunk 2]
```diff
@@ -318,7 +319,7 @@ func (s *simpleProvider) hasInvalidHostPort(container *api.Container, fldPath *f
 	allErrs := field.ErrorList{}
 	for _, cp := range container.Ports {
 		if cp.HostPort > 0 && !s.isValidHostPort(int(cp.HostPort)) {
-			detail := fmt.Sprintf("Host port %d is not allowed to be used.  Allowed ports: %v", cp.HostPort, s.psp.Spec.HostPorts)
+			detail := fmt.Sprintf("Host port %d is not allowed to be used. Allowed ports: range between {%s}", cp.HostPort, hostPortRangeToString(s.psp.Spec.HostPorts))
```
Comment (php-coder) [inline pkg/security/podsecuritypolicy/provider.go]: I'd rather use `[]` instead of `{}`

**Review by php-coder — APPROVED** [review]

File: pkg/security/podsecuritypolicy/provider.go [hunk 3]
```diff
@@ -339,3 +340,15 @@ func (s *simpleProvider) isValidHostPort(port int) bool {
 func (s *simpleProvider) GetPSPName() string {
 	return s.psp.Name
 }
+
+func hostPortRangesToString(ranges []extensions.HostPortRange) string {
+	formattedString := ""
+	if ranges != nil {
+		strRanges := []string{}
+		for _, r := range ranges {
+			strRanges = append(strRanges, fmt.Sprintf("%d-%d", r.Min, r.Max))
```
Comment (lavalamp) [inline pkg/security/podsecuritypolicy/provider.go]: If min and max are the same, this will look weird, is that allowed?

Comment (xilabao) [inline pkg/security/podsecuritypolicy/provider.go]: updated. ptal @lavalamp

**Pull request merged**